dropshot/src/error.rs

```diff
@@ -178,14 +203,16 @@ impl HttpError {
     /// `message` used for both the internal and external message.  The
     /// expectation here is that for most 400-level errors, there's no need for a
     /// separate internal message.
+    ///
+    /// The status code is provided as a [`ClientErrorStatusCode`], which may
+    /// only be constructed for 4xx client errors.
```

Comment:
I felt it was potentially helpful to state explicitly that the purpose of that
type was to ensure that this function takes only a client error, but...you're
right, the name of the function  is `for_client_error`, and the name of the type
is `ClientError...`, so that should, hopefully, be pretty obvious without this.